internal/overlord/servstate/manager.go

```diff
@@ -118,7 +133,7 @@ func (m *ServiceManager) Plan() (*plan.Plan, error) {
 // AppendLayer appends the given layer to the plan's layers and updates the
 // layer.Order field to the new order. If a layer with layer.Label already
 // exists, return an error of type *LabelExists.
-func (m *ServiceManager) AppendLayer(layer *plan.Layer) error {
+func (m *ServiceManager) AppendLayer(layer *layer.Layer) error {
```

Comment:
Hmm, yeah, `layer` is such a good (and common) variable name. This shadowing is allowed in Go, but it's a bit odd and confusing: "in this piece of code, is `layer` the package or a variable?"

Perhaps would should call the package `layers` instead? Or even `types`? (`types.Layer` reads pretty well.)